readconf.c

```diff
@@ -1734,18 +1738,140 @@ read_config_file_depth(const char *filename, struct passwd *pw,
 
 	if ((f = fopen(filename, "r")) == NULL)
 		return 0;
+	if (flags & SSHCONF_CHECKPERM) {		
+#if WINDOWS
+		/*		
+		file permissin are designed differently on windows
+		implementation on windows to make sure the config file is owned by the user of calling process
+		and nobody else except Administrators group and SYSTEM account has the write permission
+		*/
+		HANDLE h;
+		PSID owner_sid = NULL, user_sid = NULL;
+		PACL dacl = NULL;		
+		DWORD ret;		
+		PSECURITY_DESCRIPTOR pSD = NULL;		
+		char buf[2048];
+		int return_error = 0;
+		BOOL others_has_write_permission = FALSE;
+		
+		/*Get the sid of the calling process.*/
+		if ((user_sid = getusid()) == NULL) {
+			snprintf(buf, sizeof(buf), "failed to retrieve the user sid of the calling process with error: %s", strerror(errno));
+			return_error = -1;
+			goto cleanup;
+		}
+
+		/*Get the owner sid of the file.*/
+		if ((h = (HANDLE)_get_osfhandle(_fileno(f))) == INVALID_HANDLE_VALUE ||
+		    (ret = GetSecurityInfo(h, SE_FILE_OBJECT, OWNER_SECURITY_INFORMATION | DACL_SECURITY_INFORMATION,
+			&owner_sid, NULL, &dacl, NULL, &pSD)) != ERROR_SUCCESS) {
+			snprintf(buf, sizeof(buf), "failed to retrieve the owner sid of file %s", filename);
+			return_error = -1;
+			goto cleanup;
+		}
+		if ((IsValidSid(owner_sid) == FALSE) || (IsValidAcl(dacl) == FALSE)) {
+			snprintf(buf, sizeof(buf), "invalid sid or Acl is returned from  GetSecurityInfo()");
+			return_error = -1;
+			goto cleanup;
+		}
+
+		if (EqualSid(owner_sid, user_sid) == FALSE) {
+			snprintf(buf, sizeof(buf), "Bad owner on %s", filename);
+			return_error = -1;
+			goto cleanup;
+		}
+
+		/*
+		iternate all ace of the file to find out if any others than administrators group,
+		 system account, and current user account has write permission on the file
+		*/
+		for (DWORD i = 0; i < dacl->AceCount; i++)
+		{
```

Comment:
include { in the above line.. same comment for all the loops in this block below